pkg/ingestor/parser/open_vex/parser_open_vex.go

```diff
@@ -0,0 +1,152 @@
+	"github.com/guacsec/guac/pkg/ingestor/parser/common"
+	"github.com/guacsec/guac/pkg/logging"
+)
+
+var (
+	justificationsMap = map[string]generated.VexJustification{
+		"component_not_present":                             generated.VexJustificationComponentNotPresent,
+		"vulnerable_code_not_present":                       generated.VexJustificationVulnerableCodeNotPresent,
+		"vulnerable_code_not_in_execute_path":               generated.VexJustificationVulnerableCodeNotInExecutePath,
+		"vulnerable_code_cannot_be_controlled_by_adversary": generated.VexJustificationVulnerableCodeCannotBeControlledByAdversary,
+		"inline_mitigations_already_exist":                  generated.VexJustificationInlineMitigationsAlreadyExist,
+	}
+
+	vexStatusMap = map[string]generated.VexStatus{
+		"not_affected":        generated.VexStatusNotAffected,
+		"affected":            generated.VexStatusAffected,
+		"fixed":               generated.VexStatusFixed,
+		"under_investigation": generated.VexStatusUnderInvestigation,
+	}
+)
+
+type openVEXParser struct {
+	identifierStrings *common.IdentifierStrings
+	openVex           *vex.VEX
+}
+
+func NewOpenVEXParser() common.DocumentParser {
+	return &openVEXParser{
+		identifierStrings: &common.IdentifierStrings{},
+	}
+}
+
+// Parse breaks out the document into the graph components
+func (c *openVEXParser) Parse(ctx context.Context, doc *processor.Document) error {
+	return json.Unmarshal(doc.Blob, &c.openVex)
+}
+
+// GetIdentities gets the identity node from the document if they exist
+func (c *openVEXParser) GetIdentities(ctx context.Context) []common.TrustInformation {
+	return nil
+}
+
+func (c *openVEXParser) GetIdentifiers(ctx context.Context) (*common.IdentifierStrings, error) {
+	return c.identifierStrings, nil
+}
+
+func (c *openVEXParser) generateVexIngest(vulnInput *generated.VulnerabilityInputSpec, vexStatement *vex.Statement, status string) ([]assembler.VexIngest, error) {
+	var vi []assembler.VexIngest
+
+	for _, p := range vexStatement.Products {
+		vd := generated.VexStatementInputSpec{}
+		vd.KnownSince = *c.openVex.Metadata.Timestamp
+		vd.Origin = c.openVex.Metadata.ID
+
+		ingest := assembler.VexIngest{}
+
+		if vexStatus, ok := vexStatusMap[status]; ok {
+			vd.Status = vexStatus
+		}
+
+		if vd.Status == generated.VexStatusNotAffected {
+			vd.Statement = vexStatement.ImpactStatement
+		} else if vd.Status == generated.VexStatusAffected {
+			vd.Statement = vexStatement.ActionStatement
+		}
+
+		vd.VexJustification = justificationsMap[string(vexStatement.Justification)]
+
+		ingest.VexData = &vd
+		ingest.Vulnerability = vulnInput
+
+		var err error
+		if ingest.Pkg, err = helpers.PurlToPkg(p); err != nil {
+			return nil, err
+		}
+
+		c.identifierStrings.PurlStrings = append(c.identifierStrings.PurlStrings, p)
+
+		vi = append(vi, ingest)
+	}
+
+	return vi, nil
+}
+
+func (c *openVEXParser) GetPredicates(ctx context.Context) *assembler.IngestPredicates {
```

Comment:
Cut and paste the entire contents of ` GetPredicates()` (up to return statment) into `Parse()` just after the unmarshal.

Reply:
Done 👍